RadarSDK/RadarLocationManager.m

```diff
@@ -411,12 +451,82 @@ - (void)updateTrackingFromMeta:(RadarMeta *_Nullable)meta {
 
 }
 
+
+// Call this method when the tracking state changes
+- (void)changeRampedState:(BOOL)rampedUp {
+    [RadarSettings setRampedUp:rampedUp];
+    
+    StateChange *change = [[StateChange alloc] init];
+    change.timestamp = [NSDate date]; // Current timestamp
+    change.rampedUp = rampedUp;
+    [self.stateChanges addObject:change];
+}
+
+- (NSDictionary *)calculateRampedUpTimesAndCleanup {
```

Comment:
Things I'm thinking about:
1) Should we just not support a timeout in v1 to avoid the complexity of `StateChange`?
2) Can we get away with a single variable to track the running total of time spent ramped up instead of a list of `StateChange`s?
3) Should we make the timeout configurable on tracking options, something like 'nearbyTrackingTimeout`?

Reply:
Hmmm, I like the idea of having this value be configurable via the tracking options.

Note that it's less of a "timeout" and more of a "maximum allowed running time", because you can ramp up and ramp down repeatedly if you're near or transitioning through the edge of the ramp up radius.

Need to think it through a bit, but I'm pretty sure that if we can go with just a single "timeout"/"max running time", we can just store and intelligently increment a timestamp that gets compared. I'll take a stab